x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials.tsx

```diff
@@ -4,12 +4,140 @@
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { SetAppSearchChrome as SetPageChrome } from '../../../shared/kibana_chrome';
+import {
+  CredentialsLogic,
+  ICredentialsLogicActions,
+  ICredentialsLogicValues,
+} from './credentials_logic';
+import { externalUrl } from '../../../shared/enterprise_search_url/external_url';
 
 export const Credentials: React.FC = () => {
+  const { initializeCredentialsData, resetCredentials, showCredentialsForm } = useActions(
+    CredentialsLogic
+  ) as ICredentialsLogicActions;
+
+  const { dataLoading } = useValues(CredentialsLogic) as ICredentialsLogicValues;
+
+  useEffect(() => {
+    initializeCredentialsData();
+    return () => {
+      resetCredentials();
+    };
+  }, []);
+
+  // TODO
+  // if (dataLoading) { return <Loading /> }
+  if (dataLoading) {
+    return null;
+  }
   return (
     <>
-      <div>Stub</div>
+      <SetPageChrome
+        trail={[
+          i18n.translate('xpack.enterpriseSearch.appSearch.credentials.title', {
+            defaultMessage: 'Credentials',
+          }),
+        ]}
+      />
+      <EuiPageHeader>
+        <EuiPageHeaderSection>
+          <EuiTitle size="l">
+            <h1>
+              {i18n.translate('xpack.enterpriseSearch.appSearch.credentials.title', {
+                defaultMessage: 'Credentials',
+              })}
+            </h1>
+          </EuiTitle>
+        </EuiPageHeaderSection>
+      </EuiPageHeader>
+      <EuiPageContentBody>
+        <EuiFlexGroup>
+          <EuiFlexItem>
+            <EuiPanel style={{ textAlign: 'center' }}>
+              <EuiTitle size="s">
+                <h2>
+                  {i18n.translate('xpack.enterpriseSearch.appSearch.credentials.apiEndpoint', {
+                    defaultMessage: 'Endpoint',
+                  })}
+                </h2>
+              </EuiTitle>
+              <EuiCopy
+                textToCopy={externalUrl.enterpriseSearchUrl}
+                afterMessage={i18n.translate(
+                  'xpack.enterpriseSearch.appSearch.credentials.copied',
+                  {
+                    defaultMessage: 'Copied',
+                  }
+                )}
+              >
+                {(copy) => (
+                  <div>
+                    <EuiButtonIcon
+                      onClick={copy}
+                      iconType="copyClipboard"
+                      aria-label={i18n.translate(
+                        'xpack.enterpriseSearch.appSearch.credentials.copyApiEndpoint',
+                        {
+                          defaultMessage: 'Copy API Endpoint to clipboard.',
+                        }
+                      )}
+                    />
+                    <span>{externalUrl.enterpriseSearchUrl}</span>
+                  </div>
+                )}
+              </EuiCopy>
+            </EuiPanel>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        <EuiSpacer size="xxl" />
+        <EuiFlexGroup>
+          <EuiFlexItem>
```

Comment:
FYI - I think it looks/behaves a little oddly because it's meant to be used within an `<EuiPageContent>` wrapper: https://elastic.github.io/eui/#/layout/page

However, what's awkward about EuiPageContent is that it visually looks like a panel, which isn't quite what this page wants. No worries though, we can work around it, and it works out of the box on pages w/ just 1 panel (e.g. Engine Overview)

Reply:
c010b8a44e11b5e71b6946c698aaf4a0a6754506